src/WireMock.Net/RegularExpressions/RegexGuid.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace WireMock.RegularExpressions
+{
+    /// <summary>
+    /// Extension to the <see cref="Regex"/> object, adding support for GUID
+    /// tokens for matching on.
+    /// </summary>
+    public class RegexGuid : Regex
+    {
+        /// <summary>
+        /// Token for a GUID formated with `B` format specifier with lower case
+        /// values.
+        /// </summary>
+        public const string GuidBLowerToken = @"\guidb";
```

Comment:
An easier and (smaller code) approach on this would be to define a dictionary with the tokens and regex.
Like:
``` c#
private static readonly IDictionary<string,string> _replacements = new Dictionary<string, string>
{
 { @"\guidb",  @"(\{[a-z0-9]{8}-([a-z0-9]{4}-){3}[a-z0-9]{12}\})" }, // Regular expression pattern associated with the expected format for `B` format specifier with lower case.
 // . . .
}
```